lib/chef/provider/file.rb

```diff
@@ -193,6 +199,12 @@ def manage_symlink_access?
 
       private
 
+      # Return true if we are running in a docker guest, or false otherwise.
+      def docker_guest?(node)
```

Comment:
ideally i'd like to see `#docker?` pulled into Chef::DSL::Universal from chef-sugar:  

https://github.com/sethvargo/chef-sugar/blob/b9b3b643c27a0828f56d7355bb4d05577bd90a38/lib/chef/sugar/docker.rb